examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt

```diff
@@ -272,7 +272,7 @@ class SimpleMapboxNavigationKt : AppCompatActivity(), OnMapReadyCallback,
     private val routeProgressObserver = object : RouteProgressObserver {
         override fun onRouteProgressChanged(routeProgress: RouteProgress) {
             Timber.d("route progress %s", routeProgress.toString())
-            if (routeProgress.route != null) {
+            if (routeProgress.route != DirectionsRoute.builder().build()) {
```

Comment:
seems like we should handle this in `onNewRouteProgress` 🤔

Reply:
What do you mean? What's `onNewRouteProgress`?

Reply:
in NavigationMapboxMap. cause customers are going to send it too. does everyone need to check for empty routes in order to send route progress?
```
  public void onNewRouteProgress(RouteProgress routeProgress) {
    if (routeProgress.route.isEmpty()) {
       mapRoute.onNewRouteProgress(routeProgress);
    }
  }
```

Reply:
could also go into mapRoute.. should it? does it crash otherwise?

Reply:
Sync up internally, `NavigationMapRoute#onNewRouteProgress`. We should be able to remove the check from the examples 👍 as it's handled downstream 👀 

```java
val currentRoute = routeProgress.route
val hasGeometry = currentRoute?.geometry()?.isNotEmpty() ?: false
```

cc @abhishek1508 @JunDai @cafesilencio @langsmith